### ISSUE TITLE
[BUG] stream: Setting `timeout` to float does not set connect timeout in 0.6.0b9

### ISSUE THREAD

**Issue by coletdjnz** [yifeikong/curl_cffi#237]
**Describe the bug**

When using stream mode and pass a float to the timeout parameter, it only sets the read timeout and not connect timeout (CURL_OPT_CONNECTTIMEOUT_MS).

Previously in 0.5.10 it set the connect timeout and not read timeout (read timeout was addressed in https://github.com/yifeikong/curl_cffi/commit/c5f840c538c623361c80f2dce40d3c620afb3b13).

Per requests timeout handling, which curl_cffi appears to be following:
https://requests.readthedocs.io/en/latest/user/advanced/#timeouts

> If you specify a single value for the timeout, like this:
> r = requests.get('https://github.com', timeout=5)
> The timeout value will be applied to **both the connect and the read timeouts**. Specify a tuple if you would like to set the values separately:
> r = requests.get('https://github.com', timeout=(3.05, 27))

https://github.com/yifeikong/curl_cffi/blob/bf563184065e13860ee7fa9e89c7e6fbfcbff619/curl_cffi/requests/session.py#L345-L363

0.5.10: https://github.com/yifeikong/curl_cffi/blob/bb7ceba1e1bb046c0b9f53e68b3e7892f7bfa017/curl_cffi/requests/session.py#L312-L327
See last line - it sets CONNECTTIMEOUT for stream

**To Reproduce**
```py
def test_stream_connect_timeout():
    s = requests.Session()
    with pytest.raises(requests.RequestsError) as e:
        # hangs for curl default CONNECTTIMEOUT
        s.get("http://10.255.255.255", stream=True, timeout=1.0)

```

**Expected behavior**

Both read and connect timeout should be set to the float value passed to `timeout` in stream mode (with a tuple used to specify each individually).

**Versions**
 - OS: linux
 - curl_cffi==0.6.0b9
 - `pip freeze` dump
```
annotated-types==0.6.0
anyio==4.2.0
async-generator==1.10
attrs==23.2.0
autoflake==1.4
black==22.8.0
certifi==2024.2.2
cffi==1.16.0
click==8.1.7
coverage==6.4.1
cryptography==38.0.3
curl_cffi==0.6.0b9
fastapi==0.109.0
flake8==6.0.0
flake8-bugbear==22.7.1
flake8-pie==0.15.0
h11==0.14.0
httpcore==0.16.3
httpx==0.23.1
idna==3.6
iniconfig==2.0.0
isort==5.10.1
mccabe==0.7.0
mypy==0.971
mypy-extensions==1.0.0
outcome==1.3.0.post0
packaging==23.2
pathspec==0.12.1
platformdirs==4.2.0
pluggy==1.4.0
proxy.py==2.4.3
py==1.11.0
pycodestyle==2.10.0
pycparser==2.21
pydantic==2.6.0
pydantic_core==2.16.1
pyflakes==3.0.1
pytest==7.1.2
pytest-asyncio==0.19.0
pytest-trio==0.7.0
python-multipart==0.0.6
rfc3986==1.5.0
ruff==0.1.14
sniffio==1.3.0
sortedcontainers==2.4.0
starlette==0.35.1
tomli==2.0.1
trio==0.21.0
trio-typing==0.7.0
trustme==0.9.0
types-certifi==2021.10.8.2
typing_extensions==4.9.0
uvicorn==0.18.3
websockets==11.0.3
```

**Additional context**
- Which session are you using? async or sync?
- If using async session, which loop implementation are you using?

sync

**Comment by yifeikong** [comment]
Things were messed up a little bit, I will add it back.

**Comment by yifeikong** [comment]
Fixed in 0.6.0
